test/Microsoft.Azure.SignalR.Tests/ServiceConnectionFacts.cs

```diff
@@ -451,6 +452,52 @@ await proxy.WriteMessageAsync(new PingMessage()
             await serverTask3.OrTimeout();
         }
 
+        /// <summary>
+        /// If receive offline ping message and will trigger close connection
+        /// connection to default.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task CloseClientsWhenReceiveInstanceOfflinePing()
+        {
+            // Prepare clients
+            var instanceId1 = Guid.NewGuid().ToString();
+            var connectionId1 = Guid.NewGuid().ToString("N");
+            var header1 = new HeaderDictionary() { { Constants.AsrsInstanceId, instanceId1 } };
+
+            var instanceId2 = Guid.NewGuid().ToString();
+            var connectionId2 = Guid.NewGuid().ToString("N");
+            var header2 = new HeaderDictionary() { { Constants.AsrsInstanceId, instanceId2 } };
+
+            var proxy = new ServiceConnectionProxy();
+
+            var connectionTask = proxy.WaitForServerConnectionAsync(1);
+            _ = proxy.StartAsync();
+
+            await connectionTask.OrTimeout();
+
+            // 2 client connects with different instanceIds
+            await proxy.WriteMessageAsync(new OpenConnectionMessage(connectionId1, null, header1, null));
+            connectionTask = proxy.WaitForConnectionAsync(connectionId1);
+            await connectionTask.OrTimeout();
+
+            await proxy.WriteMessageAsync(new OpenConnectionMessage(connectionId2, null, header2, null));
+            connectionTask = proxy.WaitForConnectionAsync(connectionId2);
+            await connectionTask.OrTimeout();
+
+            // Server received instance offline ping on instanceId1 and trigger cleanup related client1
+            await proxy.WriteMessageAsync(new PingMessage()
+            {
+                Messages = new[] { "offline", instanceId1 }
+            });
+
+            // Validate client1 is closed and client2 is still connected
+            var disconnectTask = proxy.WaitForConnectionCloseAsync(connectionId1);
+            await disconnectTask.OrTimeout();

```

Comment:
>disconnectTask [](start = 18, length = 14)

any test to validate that disconnect task is handled?

Reply:
current validation not enough? what else shall we check?

Reply:
I don't think this one checks the hub's disconnect event.

---
In reply to: [307564663](https://github.com/Azure/azure-signalr/pull/612#discussion_r307564663) [](ancestors = 307564663)